pkg/cache/cache.go

```diff
@@ -312,13 +317,45 @@ func (c *Cache) DeleteResourceFlavor(rf *kueue.ResourceFlavor) sets.String {
 }
 
 func (c *Cache) ClusterQueueActive(name string) bool {
+	return c.clusterQueueInStatus(name, Active)
+}
+
+func (c *Cache) ClusterQueueTerminating(name string) bool {
```

Comment:
I agree. Moreover, we are checking if the cluster queue is inactive and then setting it to inactive using a separate lock. This could lead to race conditions. We should be using one lock.

Reply:
The race condition here doesn't matter much though, `TerminateClusterQueue` should be idempotent. I think Kante just wanted to avoid to always acquire the write lock.

Reply:
Yes, exactly. Always requiring a R&W lock is too expensive, but there's a gap between release the read lock and require a R&W lock.

Reply:
Right, a terminating CQ would remain terminating until fully deleted, so it doesn't matter. Sounds good.